contracts/truefi2/SAFU.sol

```diff
@@ -142,7 +142,8 @@ contract SAFU is ISAFU, UpgradeableClaimable {
      * @param amount Amount of deficiency tokens to be reclaimed
      */
     function reclaim(IDebtToken loan, uint256 amount) external override {
-        require(loanFactory.isLoanToken(loan), "SAFU: Unknown loan");
+        require(loanFactory.isCreatedByFactory(loan), "SAFU: Unknown loan");
```

Comment:
Are we sure if we want this? Later on we call:
`IDeficiencyToken dToken = deficiencyToken[loan];`
and a few lines after that:
`dToken.burnFrom(msg.sender, amount);`

I'm pretty sure it will revert any time we provide a debt token as an argument anyway.